archiver.go

```diff
@@ -128,34 +129,33 @@ func (a *Archiver) do(log *logrus.Entry, j *Job) (err error) {
 		}
 	}()
 
-	log.WithFields(logrus.Fields{
+	l.New(log.Fields{
 		"status":     r.Status,
 		"last-fetch": r.FetchedAt,
 		"references": len(r.References),
-	}).Debug("repository model obtained")
+	}).Debugf("repository model obtained")
 
 	defer func() {
 		if rcv := recover(); rcv != nil {
-			log.WithField("error", rcv).Error("panic while processing repository")
-
+			l.Error(fmt.Errorf("%v", rcv), "panic while processing repository")
 			r.FetchErrorAt = &now
-			if sErr := a.Store.UpdateFailed(r, model.Pending); sErr != nil {
-				log.WithFields(logrus.Fields{
+			if err := a.Store.UpdateFailed(r, model.Pending); err != nil {
+				l.New(log.Fields{
 					"err": err,
 					"id":  r.ID,
-				}).Error("error setting repo as failed")
+				}).Error(err, "error setting repo as failed")
 			}
 
 			err = ErrFatal.New(rcv, debug.Stack())
 		}
 	}()
 
 	if err := a.canProcessRepository(r, &now); err != nil {
-		log.WithFields(logrus.Fields{
+		l.New(log.Fields{
```

Comment:
This is not the intended use for `New`. 
New is expected to be used for pass a `logger` to the context, and not to directly inline.